sdk/communication/azure-communication-common-perf/src/main/java/com.azure.communication.common.perf/HmacAuthenticationPolicyTest.java

```diff
@@ -47,21 +47,24 @@ public HmacAuthenticationPolicyTest(PerfStressOptions options) throws MalformedU
     @Override
     public void run() {
         HttpResponse response = pipeline.send(request).block();
-        String date = response.getRequest().getHeaders().getValue("date");
+        String date = response.getRequest().getHeaders().getValue("x-ms-date");
         String signature = response.getRequest().getHeaders().getValue("Authorization");
         checkSignatureCorrectness(date, signature);
     }
 
     @Override
     public Mono<Void> runAsync() {
-        HttpResponse response = pipeline.send(request).block();
-        //getting date header from old implementation
-        //String date = response.getRequest().getHeaders().getValue("date");
-        //getting date header from new implementation
-        String date = response.getRequest().getHeaders().getValue("x-ms-date");
-        String signature = response.getRequest().getHeaders().getValue("Authorization");
-        checkSignatureCorrectness(date, signature);
-        return Mono.empty();
+        return pipeline.send(request)
+            .map(response -> {
+                String date = response.getRequest().getHeaders().getValue("x-ms-date");
+                String signature = response.getRequest().getHeaders().getValue("Authorization");
+                try {
+                    checkSignatureCorrectness(date, signature);
+                } catch (Exception e) {
+                    return Mono.error(e);
+                }
+                return Mono.empty();
+            }).then();
```

Comment:
This hides potential exception thrown in map block. You have to either remove try-catch block or use flatMap instead of map.

Reply:
but on the 64th line, we are propagating the actual error

Reply:
Not really. The result of pipeline.send().map(..) is of type Mono<Mono<...>> and the actual error is in the inner Mono but the outer Mono contains "some" object which happens to be another mono with the error.

Reply:
That's why you need to use `flatMap `- to unwrap the "outer" mono.

Reply:
Oh, I understood your point, will fix it